dot/sync/chain_sync.go

```diff
@@ -88,7 +88,7 @@ type workHandler interface {
 	handleTick() ([]*worker, error)
 }
 
-//go:generate mockgen -destination=mock_chain_sync_test.go -package $GOPACKAGE . ChainSync
+//go:generate mockgen -destination=mock_chain_sync_test.go -package=$GOPACKAGE -source=chain_sync.go . ChainSync,workHandler
```

Comment:
Updated.

Reply:
This change didn't work because it generates ChianSync in both files.

Reply:
Ugh actually specifying `workHandler` when using mockgen in source mode is useless, it will just generate mocks for all interfaces found in `chain_sync.go`. Maybe having work handler in another Go file would do in this case. Or leave as is, up to you, it's not a big problem anyway.

Reply:
Yes, it's frustrating, I'm leaving it for now.